pkg/controller/jobframework/reconciler.go

```diff
@@ -365,7 +366,24 @@ func (r *JobReconciler) equivalentToWorkload(job GenericJob, object client.Objec
 	if owner.Name != object.GetName() {
 		return false
 	}
-	return job.EquivalentToWorkload(*wl)
+
+	jobPodSets := job.PodSets()
+	if job.IsSuspended() {
```

Comment:
The main purpose is to decrees the complexity of the `GenericJob` interface, not so much for the Jobs
we have integrated now but for the potential new ones.

`jobPodSets[i].Count > wl.Spec.PodSets[i].Count` is to ensure that once the workload is admitted, the job will not try to use more resources that  it has initially requested.

I have tried to improve the comments.

Reply:
Apologize for the delay ... 

So for unsuspended Job, we will not compare the count right? I think this is a bit different than what we're today(for mpiJob and rayJob), isn't that should be counted?

Reply:
We are checking that it's not using more that what known during admission. I the future , if we are 100% sure that the job is not supporting Partial admission  or dynamic reclaim we can make a strict check.

For now the trend is to try to add support for partial-admission  and dynamic-reclaim in all the frameworks.

Reply:
em... actually what I learned in the past is we should **be strict at first** then evolve based on the feedbacks, but we're doing the opposite, we change the current behavior based on our assumptions. 

Or we can add a new method to the job interface to know whether the Job supports partial admission, but it seems more complexities then. Glad to hear other's advices here since it's a tradeoff. cc @alculquicondor @tenzen-y

Reply:
> My suggestion is leave the interface equivalentToWorkload as it is, and make functions like ComparePodSetSlices as util ones.

+1 with @kerthcet 

@alculquicondor @kerthcet @trasc PartialAdmission is an alpha-stage feature. So once the feature graduates to GA, can we re-evaluate whether removing `EquivalentToWorkload ` from `GenericJob` interface is worth it?

Reply:
I've updated this, the count check is skipped only if the job is unsuspended and it accepts partial admission, in which case only the podSets with variable count can have a different count which should be
in the  minCout .. Count interval.  The check is as **strict** as before for all the integrations.

If in the future, we need a more specific approach for some frameworks we can add an optional interface then.